Example/Controllers/SwiftUI/Features/List/MiniAppListViewModel.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+import MiniApp
+import Combine
+
+@MainActor
+class MiniAppListViewModel: ObservableObject {
+
+    let store = MiniAppStore.shared
+
+    @Published var type: MiniAppSettingsView.ListConfig
+    @Published var state: ListState = .none
+    @Published var indexedMiniAppInfoList: [String: [MiniAppInfo]] = [:]
+
+    var bag = Set<AnyCancellable>()
+
+    init(type: MiniAppSettingsView.ListConfig) {
+        self.type = type
+        setupObservers()
+        if store.miniAppSetupCompleted {
+            load()
+        } else {
+            state = .awaitsSetup
+        }
+    }
+
+    func checkInitialLoad() {
+        // checkSetup()
+        // if store.miniAppSetupCompleted && indexedMiniAppInfoList.isEmpty {
+        //     load()
+        // }
```

Comment:
Do we need this method ?

Reply:
Currently not maybe later for cleanup. I can remove it for now.